package/samplers/value_at_risk/_gp/acqf.py

```diff
@@ -233,10 +283,25 @@ def eval_acqf(self, x: torch.Tensor) -> torch.Tensor:
 
         if self._acqf_type == "mean":
             return self._value_at_risk(x).mean(dim=-1)
-        elif self._acqf_type == "nei":
-            raise NotImplementedError("NEI is not implemented yet.")
-        else:
-            raise ValueError(f"Unknown acqf_type: {self._acqf_type}")
+
+        x_noisy = x.unsqueeze(-2) + self._input_noise
+        means, covar = self._gpr.posterior(x_noisy, joint=True)
+        cov21 = self.posterior_covar_cross_block(x_noisy)
+        L21 = torch.linalg.solve_triangular(
+            self._covar_robust_chol.transpose(-1, -2), cov21, left=False, upper=True
+        )
+        # Add a small jitter to prevent the matrix from becoming non-positive definite due to numerical issues.
+        # This issue appears to happen when the x is included in robust_X.
+        L22 = torch.linalg.cholesky(
+            covar - L21.matmul(L21.transpose(-1, -2)) + 1e-12 * torch.eye(covar.shape[-1])
+        )
+        posterior_at_x_noisy = (
+            means.unsqueeze(-2)
+            + torch.einsum("...ij,kj->ki", L21, self._S1)
+            + self._S2.matmul(L22)
+        )
```

Comment:
The batch direction is removed in the second term, but it's recovered by third term. How about not removing in the second term?

```suggestion
        posterior_at_x_noisy = (
            means.unsqueeze(-2)
            + torch.einsum("...ij,kj->...ki", L21, self._S1)
            + self._S2.matmul(L22)
        )
```

Reply:
Great catch! I believe this is a serious bug.